### DISCUSSION TITLE
#each with limit, not using index

### DISCUSSION THREAD

**Discussion by Parmstig** [microsoft-search/pnp-modern-search#2511] [Q&A] [answered]
I want to use the full widht section and show 6 cards in one row.
3 TopNews | 3 Regular News

My filter is working but i cant limit array data.items to 3.

`        <div class="template--topnews">
            {{#each data.items as |item|}}
                {{#eq item.TopNewsOWSBOOL '0'}}
                    <div>{{TopNewsOWSBOOL}} - {{Title}}</div>
                {{/eq}}
            {{/each}}
        </div>`
Is there a built in helper for limiting each to just return the first three. I have tried with index but since the index returned from the whole array that will not solve problem

A better way would be to filter with {{#each data.item WHERE item.TopNewsOWSBOOL '0'}} but i dont know how to do that.

**Comment by wobba** [comment, marked as answer]
You can try  https://github.com/helpers/handlebars-helpers#before or https://github.com/helpers/handlebars-helpers#filter

**Reply by Parmstig** [reply]
thx wobba
